.github/workflows/pimcore-upgrade-demo.yml

```diff
@@ -0,0 +1,120 @@
+        #
+        # See: https://getcomposer.org/doc/03-cli.md#create-project
+        # See: https://getcomposer.org/doc/05-repositories.md#path
+        docker run \
+          --volume=${{ github.workspace }}/:/test/ \
+          --workdir=/test/ \
+          --user=$(id -u):$(id -g) \
+          docker.io/pimcore/pimcore:php8.2-latest \
+            composer create-project \
+              pimcore/demo:@dev \
+              --repository='{"type": "path", "url": "./demo"}' \
+              sample-project
+
+    - name: Smoke-test compose file
+      run: |
+        # Echo commands and terminate on first error
+        set -ex
+        
+        # Check (lint) the compose file
+        docker-compose -v
+        cd sample-project/
+        docker-compose config
+
+    - name: Test pimcore installation
+      run: |
+        # Echo commands and terminate on first error
+        set -ex
+
+        cd sample-project/
+
+        # Start containers
+        docker-compose pull --quiet
+        docker-compose down -v --remove-orphans
+        docker-compose up -d
+
+        # Install dockerize into the php container. We need it to block until
+        # database is ready to serve connections.
+        docker-compose exec -T -- php bash -c '\
+          curl -sfL https://github.com/powerman/dockerize/releases/download/v0.11.5/dockerize-`uname -s`-`uname -m` \
+          | install /dev/stdin /usr/local/bin/dockerize'
+
+        # Wait for the database to set up.
+        docker-compose exec -T -- php dockerize -wait tcp://db:3306 -timeout 5m
+          
+        # Run pimcore installation.
+        docker-compose exec -T \
+        -e PIMCORE_INSTALL_ADMIN_USERNAME=pimcore \
+        -e PIMCORE_INSTALL_ADMIN_PASSWORD=pimcore \
+        -e PIMCORE_INSTALL_MYSQL_USERNAME=pimcore \
+        -e PIMCORE_INSTALL_MYSQL_PASSWORD=pimcore \
+        -- \
+        php vendor/bin/pimcore-install -n --mysql-host-socket=db --mysql-database=pimcore
+          
+        # Change owner  
+        sudo chown -R www-data .  
+        
+        # Check if website is reachable
+        response=$(docker-compose exec -T -- php bash -c 'curl -s "nginx:80"')
+        
+        if [[ ! $response =~ "Satisfaction" ]]; then
+           echo "Install failed, skipping build"
+           exit 1;
+        fi
+        
+        # Upgrade to latest dev branch and re-test  
+        docker-compose exec -T -- php rm composer.lock
+        docker-compose exec -T -- php composer config minimum-stability dev
+        docker-compose exec -T -- php composer require -n "pimcore/pimcore:11.x-dev as 11.99.99"  
+        
+        # Check if website with latest dev version is reachable
+        response=$(docker-compose exec -T -- php bash -c 'curl -s "nginx:80"')
+      
+        if [[ ! $response =~ "Satisfaction" ]]; then
+           echo "Install upgrade failed, skipping build"
+           exit 2;
+        fi
+          
```

Comment:
NB: works on 11.0 because of https://github.com/pimcore/demo/pull/516 but on 11.x is not applied